### ISSUE TITLE
official apt repo?

### ISSUE THREAD

**Issue by josephholsten** [python-diamond/Diamond#222]
It's super easy to build a deb from diamond, but I imagine most users would be happy just installing from an existing repo.

[Bintray](https://bintray.com/) & [packagecloud](https://packagecloud.io/) both have free hosting for OSS projects. Would anyone object to me adding that to travis and the release process?

**Comment by tiadobatima** [comment]
+1000000000

Would love to see diamond in an official repo. It would help adoption considerably.

@liquidgecka has done some work on this already:
- https://github.com/BrightcoveOS/Diamond/issues/187#issuecomment-50942306
- https://launchpad.net/~orchestrate/+archive/ubuntu/standard

**Comment by liquidgecka** [comment]
I could very easily make this something that the maintainers could push if so desired.. Setting up a launchpad account and group is fairly easy, and pushing new deploys is actually quite easy. If desired I could set a PR with a script that can be run by a maintainer to deploy a release to a ppa.

The script I currently use is here: https://gist.github.com/liquidgecka/c03fb64b1dc309995d2d though for this to be official it would ideally just use the existing debian/changelog file (which could easily be modified to pull the history since the last release or something of that sort) and a real version (at the moment it uses a timestamp + git sha).

If there is real interest from the maintainers of Diamond let me know. I can script this out and make it super simple so its not hard to cut a release and if there is already a release process then I could help tap into that so it requires no new steps. (I have actually set this up as a travis job before as well though that requires a bunch of secure environment variables.) =)

**Comment by jaingaurav** [comment]
@liquidgecka I think there is a strong desire for this along with regular pip releases. If you can help get some of the frameworks/scripts in place it would be much appreciated.

**Comment by liquidgecka** [comment]
So if the intention is to use ubuntu ppa system then this becomes _really_ easy.. It could be easily scripted and added to Travis.

1) I just requested ownership of the 'diamond' team in ubuntu launchpad (its been abandoned since 2011), but if taking over that name is not possible we would just need to pick a team name and add a few members/maintainers.
2) The next step would be to add debian packaging information that uses the git helpers that nicely wrap up git commits and all that. I can do that in a pull request fairly easily.
3) The last step is to add a dput command to travis (and some secure environment variables) and after that it would automatically build and release when travis runs.

Do you have a launchpad account already? If so you should make one so a team can get setup and all that.

**Comment by jaingaurav** [comment]
Yes I have a launchpad account with id 'gaurav'

Where is the old diamond team in ubuntu? Maybe we can ping some of the people. If you have any issues, you could just use 'python-diamond', probably less confusing than a generic 'diamond'

Thanks for you help here, I know nothing about ppa

**Comment by liquidgecka** [comment]
I already tried contacting them.. we will have to wait and see if they respond. Otherwise I created a python-diamond team and added you as an administrator.

**Comment by jaingaurav** [comment]
Yeah let's stick with python-diamond for the team name to keep consistency with github. If we get ownership of the legacy team we can simply delete it.

**Comment by liquidgecka** [comment]
Sorry, this is taking longer than expected (my wife and I just had a baby so things have gotten chaotic..). I am still working on it though. One thing that will make this easier: is there an official "release" script/process? Debian packaging uses a stupid system for change log management that requires things to be committed to the repo in most cases. I wanted to script this out so it would be automated, but that only works if I can tell what version a release is.. etc.

**Comment by liquidgecka** [comment]
Sorry, this is taking longer than expected (my wife and I just had a baby so things have gotten chaotic..). I am still working on it though. One thing that will make this easier: is there an official "release" script/process? Debian packaging uses a stupid system for change log management that requires things to be committed to the repo in most cases. I wanted to script this out so it would be automated, but that only works if I can tell what version a release is.. etc.

**Comment by liquidgecka** [comment]
Sorry, this is taking longer than expected (my wife and I just had a baby so things have gotten chaotic..). I am still working on it though. One thing that will make this easier: is there an official "release" script/process? Debian packaging uses a stupid system for change log management that requires things to be committed to the repo in most cases. I wanted to script this out so it would be automated, but that only works if I can tell what version a release is.. etc.

**Comment by jaingaurav** [comment]
Congratulations and thanks for all your help!

I'm not too sure about the official release process, however the version number is based on the number of commits since the 4.0 release. I'd like to get out a new release in the next month or so so hopefully we'll figure out the kinks while doing that.

**Comment by liquidgecka** [comment]
Okay so new baby and a work trip have managed to kick my butt pretty effectively. =)

So how about this.. Is it possible to add a script to the release process that prepends some lines into debian/changelog and then commits that? If we can do that then I can easily setup travis to build every time that file changes (which is kind of the debain 'way') and if its included in the release script then it would basically update on release. The downside is that it requires a commit to the repo to make it all work.

**Comment by jruere** [comment]
This would be a pretty neat addition, if completed. :)

**Comment by DStape** [comment]
It should be in Debian 9: https://packages.debian.org/stretch/diamond
